src/LinAlg/hiopVector.hpp

```diff
@@ -198,6 +198,13 @@ class hiopVector
   virtual long long get_local_size() const = 0;
   virtual double* local_data() = 0;
   virtual const double* local_data_const() const = 0;
+  virtual double* local_data_host() = 0;
+  virtual const double* local_data_host_const() const = 0;
+
+  virtual void copyToDev() = 0;
+  virtual void copyFromDev() = 0;
+  virtual void copyToDev() const = 0;
+  virtual void copyFromDev() const = 0;
 
 protected:
   long long n_; //we assume sequential data
```

Comment:
what's the difference between class hiopVector and class hiopVectorIntSeq. Seems both of them assume sequential data, and length of vector is a given value from initialization. The only difference seems to me is that hiopVector asks for more pure functions. Can we link hiopVector with hiopVectorInt or hiopVectorIntSeq?

Reply:
`hiopVectorInt` is a base class for integer vectors. They are supposed to be used for mapping equality and inequality constraints. For now they are not used in the code, but will be needed to enable computations in the device memory instead of unified virtual memory.